R/lazyframe-frame.R

```diff
@@ -2610,4 +2610,185 @@ lazyframe__sink_csv <- function(
   })
 }
 
+#' Creates a summary of statistics for a LazyFrame, returning a DataFrame.
+#'
+#' @description
+#' This method does not maintain the laziness of the frame, and will collect
+#' the final result. This could potentially be an expensive operation.
+#'
+#' We do not guarantee the output of `describe()` to be stable. It will show
+#' statistics that we deem informative, and may be updated in the future. Using
+#' `describe()` programmatically (versus interactive exploration) is not
+#' recommended for this reason.
+#'
+#' @param percentiles One or more percentiles to include in the summary
+#' statistics. All values must be in the range `[0; 1]`.
+#' @param interpolation Interpolation method for computing quantiles. Must be
+#' one of `"nearest"`, `"higher"`, `"lower"`, `"midpoint"`, or `"linear"`.
+#'
+#' @details
+#' The median is included by default as the 50% percentile.
+#'
+#' @inherit as_polars_df return
+#' @examples
+#' lf <- pl$LazyFrame(
+#'   int = 1:3,
+#'   float = c(0.5, NA, 2.5),
+#'   string = c(letters[1:2], NA),
+#'   date = c(as.Date("2024-01-20"), as.Date("2024-01-21"), NA),
+#'   cat = factor(c(letters[1:2], NA)),
+#'   bool = c(TRUE, FALSE, NA)
+#' )
+#' lf$collect()
+#'
+#' # Show default frame statistics:
+#' lf$describe()
+#'
+#' # Customize which percentiles are displayed, applying linear interpolation:
+#' lf$describe(
+#'   percentiles = c(0.1, 0.3, 0.5, 0.7, 0.9),
+#'   interpolation = "linear"
+#' )
+lazyframe__describe <- function(
+  percentiles = c(0.25, 0.5, 0.75),
+  interpolation = c("nearest", "higher", "lower", "midpoint", "linear")
+) {
+  wrap({
+    schema <- self$collect_schema()
+    if (length(schema) == 0) {
+      abort("cannot describe a LazyFrame without any columns")
+    }
+
+    # create list of metrics
+    metrics <- c("count", "null_count", "mean", "std", "min")
+    quantiles <- parse_percentiles(percentiles)
+    if (length(quantiles) > 0) {
+      metrics <- c(metrics, paste0(percentiles * 100, "%"))
+    }
+    metrics <- c(metrics, "max")
+
+    skip_minmax <- function(x) {
+      # TODO: need to check if x is object, or unknown
```

Comment:
Can we have a datatype object or unknown?

Reply:
All types are listed here; Unknown exists but Object does not.

https://github.com/eitsupi/neo-r-polars/blob/c053c0f9f8677c67356839bf4bebb7156747548d/src/rust/src/datatypes.rs#L263-L293

Reply:
> we have any examples with this.

Indeed, there is no exposed way to create an instance of this type.

```r
dt <- PlRDataType$new_from_name("Unknown") |> wrap()
dt
#> Unknown(Any)
class(dt)
#> [1] "polars_dtype_unknown" "polars_dtype"         "polars_object" 

```